pydantic/typing.py

```diff
@@ -397,3 +405,12 @@ def get_class(type_: Type[Any]) -> Union[None, bool, Type[Any]]:
     except (AttributeError, TypeError):
         pass
     return None
+
+
+def get_sub_types(tp: Any) -> List[Any]:
+    if get_origin(tp) is Annotated:
+        return get_sub_types(get_args(tp)[0])
+    elif get_origin(tp) is Union:
+        return [x for t in get_args(tp) for x in get_sub_types(t)]
```

Comment:
```suggestion
    origin = get_origin(tp)
    if origin is Annotated:
        return get_sub_types(get_args(tp)[0])
    elif origin is Union:
        return [x for t in get_args(tp) for x in get_sub_types(t)]
```

Also, please add a docstring.